### PR TITLE
Clean Up hyundai

### PR DESCRIPTION
**Pull request by emmertex**
This is one of the first parts of upstreaming new models, features and tuning.

Alphabetical Order for Cars and Fingerprints
Generic DBC
Credit users who have made each vehicle possible.

Requires
https://github.com/commaai/opendbc/pull/130

### CODE REVIEW

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -11,11 +11,11 @@ def get_hud_alerts(visual_alert, audible_alert):
     return 0
 
 class CAR:
-  SANTA_FE = "HYUNDAI SANTA FE LIMITED 2019"
-  ELANTRA = "HYUNDAI ELANTRA LIMITED ULTIMATE 2017"
-  GENESIS = "HYUNDAI GENESIS 2018"
-  KIA_SORENTO = "KIA SORENTO GT LINE 2018"  # Top Trim Kia Sorento for Australian Market, AWD Diesel 8sp Auto
-  KIA_STINGER = "KIA STINGER GT2 2018"
+  ELANTRA = "HYUNDAI ELANTRA LIMITED ULTIMATE 2017"   # First User @TK211X    <-- Ported by @ku7 (Second, First Hyundai in World)
```
Comment (vanillagorillaa) [inline selfdrive/car/hyundai/values.py]: What's the reason for all these changes?

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -39,6 +36,9 @@ class Buttons:
   CAR.KIA_STINGER: [{
     67: 8, 127: 8, 304: 8, 320: 8, 339: 8, 356: 4, 358: 6, 359: 8, 544: 8, 576: 8, 593: 8, 608: 8, 688: 5, 809: 8, 832: 8, 854: 7, 870: 7, 871: 8, 872: 8, 897: 8, 902: 8, 909: 8, 916: 8, 1040: 8, 1042: 8, 1056: 8, 1057: 8, 1064: 8, 1078: 4, 1107: 5, 1136: 8, 1151: 6, 1168: 7, 1170: 8, 1173: 8, 1184: 8, 1265: 4, 1280: 1, 1281: 4, 1287: 4, 1290: 8, 1292: 8, 1294: 8, 1312: 8, 1322: 8, 1342: 6, 1345: 8, 1348: 8, 1363: 8, 1369: 8, 1378: 4, 1379: 8, 1384: 8, 1407: 8, 1419: 8, 1425: 2, 1427: 6, 1456: 4, 1470: 8
   }],
+  CAR.SANTA_FE: [{
```
Comment (vanillagorillaa) [inline selfdrive/car/hyundai/values.py]: This one too?

File: selfdrive/car/hyundai/values.py [hunk 3]
```diff
@@ -50,9 +50,9 @@ class Buttons:
 }
 
 DBC = {
-  CAR.SANTA_FE: dbc_dict('hyundai_santa_fe_2019_ccan', None),
-  CAR.ELANTRA: dbc_dict('hyundai_santa_fe_2019_ccan', None),
-  CAR.GENESIS: dbc_dict('hyundai_santa_fe_2019_ccan', None),
-  CAR.KIA_SORENTO: dbc_dict('hyundai_santa_fe_2019_ccan', None),
-  CAR.KIA_STINGER: dbc_dict('hyundai_santa_fe_2019_ccan', None),
+  CAR.ELANTRA: dbc_dict('hyundai_kia_generic', None),
```
Comment (vanillagorillaa) [inline selfdrive/car/hyundai/values.py]: Seems like these changes are really the only ones that are needed

File: selfdrive/car/hyundai/values.py (hunk 2 above)
Comment (rbiasini) [reply selfdrive/car/hyundai/values.py]: Alphabetical order. I think these changes are ok. Not sure about the dbc file change below though. Is the kia one a superset of the santa fe?

Comment (emmertex) [reply selfdrive/car/hyundai/values.py]: There's no reason to add ambiguity into the naming of the dbc.
All Hyundai and Kia use the same DBC and it has absolutely nothing to do specifically with any model, and nothing at all to do specifically with santa Fe.
Easier to name things properly now.

Comment (emmertex) [reply selfdrive/car/hyundai/values.py]: Historically it was all from the leaked dbc, I added a fix to an error in a message, and then you added a message regarding gear selection.
Both of these changes are true and valid for every vehicle encountered.

Comment (rbiasini) [reply selfdrive/car/hyundai/values.py]: ok. merged the opendbc repo. before I can merge this, you should update your openpilot repo with the latest dbc files.

File: selfdrive/car/hyundai/values.py (hunk 1 above)
Comment (emmertex) [reply selfdrive/car/hyundai/values.py]: done

**Comment by rbiasini** [comment]
Looks good, will merge. Thanks

**Pull request merged**